### ISSUE TITLE
Takes 5 seconds for keyboard to come up

### ISSUE THREAD

**Issue by sigpoggy** [hackiftekhar/IQKeyboardManager#172]
When testing on a first gen iPad mini and 3rd gen iPad (retina) it takes 5 seconds for the keyboard to come up after tapping a text field. This only happens once, the first time a field is tapped after starting up the app. There is no timing problem after that.

If you can't find a solution, can you find a work-around like doing the first time startup code in the background before any field is accessed.

**Comment by sigpoggy** [comment]
This may help. Note the 5 second delay between textFieldViewDidBeginEditing: ended and keyboardWillShow: started.

2015-02-19 16:22:39.755 IQKeyboardManager[5129:2685445] IQKeyboardManager: ****** textFieldViewDidBeginEditing: ended ******
2015-02-19 16:22:44.681 IQKeyboardManager[5129:2685445] IQKeyboardManager: ****** keyboardWillShow: started ******

**Comment by mythodeia** [comment]
@sigpoggy your tests are on devices or on simulators?

**Comment by sigpoggy** [comment]
devices

**Comment by mythodeia** [comment]
did you try the demo project as well or your own?

**Comment by theothertomelliott** [comment]
This might not be the same issue, but I've been seeing something similar on my own app. If you remove IQKeyboardManager, do you still see the same thing?

If so, it might be due to a segue that is happening on a thread other than main.

**Comment by sigpoggy** [comment]
yes
The log message I sent showing the timing is from the demo app

**Comment by sigpoggy** [comment]
This appears to be an Apple bug. I created a test app than only has a text field - no IQKeyboardManager code - and checked the timing between the text field did begin editing and the keyboard did show and it always took about five seconds the first time (ipad mini)
One workaround I came up with involves adding some code in the app didFinishLaunchingWithOptions function:
1. add self as observer to keyboard will show (add function to handle keyboard will show notification)
2. add a temporary text field to the window.rootViewController
3. make it become firstRespnder
4. In the keyboard will show notification function, resign the text field as firstResponder, remove from superview, remove self as keyboard observer.

A side effect of this is that startup is delayed the first time launched, but the launch screen is showing so the user assumes all is going as planned.
